hazelcast/src/main/java/com/hazelcast/config/ConfigYamlGenerator.java

```diff
@@ -0,0 +1,1266 @@
+            addNonNullToMap(subConfigAsMap, "merge-policy",
+                    getMergePolicyConfigAsMap(subConfigAsObject.getMergePolicyConfig()));
+
+            child.put(subConfigAsObject.getName(), subConfigAsMap);
+        }
+
+        parent.put("cardinality-estimator", child);
+    }
+
+    static void flakeIdGeneratorYamlGenerator(Map<String, Object> parent, Config config) {
+        if (config.getFlakeIdGeneratorConfigs().isEmpty()) {
+            return;
+        }
+
+        Map<String, Object> child = new LinkedHashMap<>();
+        for (FlakeIdGeneratorConfig subConfigAsObject : config.getFlakeIdGeneratorConfigs().values()) {
+            Map<String, Object> subConfigAsMap = new LinkedHashMap<>();
+
+            addNonNullToMap(subConfigAsMap, "prefetch-count",
+                    subConfigAsObject.getPrefetchCount());
+            addNonNullToMap(subConfigAsMap, "prefetch-validity-millis",
+                    subConfigAsObject.getPrefetchValidityMillis());
+            addNonNullToMap(subConfigAsMap, "epoch-start",
+                    subConfigAsObject.getEpochStart());
+            addNonNullToMap(subConfigAsMap, "node-id-offset",
+                    subConfigAsObject.getNodeIdOffset());
+            addNonNullToMap(subConfigAsMap, "bits-sequence",
+                    subConfigAsObject.getBitsSequence());
+            addNonNullToMap(subConfigAsMap, "bits-node-id",
+                    subConfigAsObject.getBitsNodeId());
+            addNonNullToMap(subConfigAsMap, "allowed-future-millis",
+                    subConfigAsObject.getAllowedFutureMillis());
+            addNonNullToMap(subConfigAsMap, "statistics-enabled",
+                    subConfigAsObject.isStatisticsEnabled());
+
+            child.put(subConfigAsObject.getName(), subConfigAsMap);
+        }
+
+        parent.put("flake-id-generator", child);
+    }
+
+    static void pnCounterYamlGenerator(Map<String, Object> parent, Config config) {
+        if (config.getPNCounterConfigs().isEmpty()) {
+            return;
+        }
+
+        Map<String, Object> child = new LinkedHashMap<>();
+        for (PNCounterConfig subConfigAsObject : config.getPNCounterConfigs().values()) {
+            Map<String, Object> subConfigAsMap = new LinkedHashMap<>();
+
+            addNonNullToMap(subConfigAsMap, "replica-count",
+                    subConfigAsObject.getReplicaCount());
+            addNonNullToMap(subConfigAsMap, "split-brain-protection-ref",
+                    subConfigAsObject.getSplitBrainProtectionName());
+            addNonNullToMap(subConfigAsMap, "statistics-enabled",
+                    subConfigAsObject.isStatisticsEnabled());
+
+            child.put(subConfigAsObject.getName(), subConfigAsMap);
+        }
+
+        parent.put("pn-counter", child);
+    }
+
+    static void wanReplicationYamlGenerator(Map<String, Object> parent, Config config) {
+        if (config.getWanReplicationConfigs().isEmpty()) {
```

Comment:
just fyi: currently `addWanReplicationConfig` is not supported. It was not meant to be supported in the original PR https://github.com/hazelcast/hazelcast/pull/10757 but instead of throwing an `UnsupportedOperationException` it just added the provided config to the list of static configs (which doesn't do much anyway).
However there is a way to dynamically add WAN config (either via REST API or via management center), so probably it makes sense to eventually add support for that via programmatic config API as well. 
I will create a separate issue for that, you can leave the YAML generator as is in this PR.

Reply:
To add, I didn't finish integrating everything (configuration options must be implemented), but in my dev branch I made dynamic wan changes persisted. https://github.com/ramizdundar/hazelcast-enterprise/commit/30bdcdbc340d75b2897b9021a3c0e5f662db5219